packages/api/src/command/medical/document/document-query.ts

```diff
@@ -210,18 +207,20 @@ export const updateConversionProgress = async ({
  * Returns the existing request ID if the previous query has not been entirely completed. Otherwise, returns a newly-generated request ID.
  *
  * @param docQueryProgress Progress of the previous query
+ * @param forceNew Force creating a new request ID
  * @returns uuidv7 string ID for the request
  */
-function getOrGenerateRequestId(docQueryProgress: DocumentQueryProgress | undefined): string {
-  if (!docQueryProgress) return uuidv7();
+function getOrGenerateRequestId(
+  docQueryProgress: DocumentQueryProgress | undefined,
+  forceNew = false
+): string {
+  if (forceNew) return generateRequestId();
 
-  const isDownloadFinished = docQueryProgress.download?.status === "completed";
-  const conversionStatus = docQueryProgress.convert?.status ?? undefined;
```

Comment:
This is the reason to start working on this PR: if the status was `failed` we wouldn't generate new request IDs.